segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -252,11 +254,19 @@ public void initialize(long containerEpoch) {
 
                                     // This instance is the owner, return a handle.
                                     val retValue = SegmentStorageHandle.writeHandle(streamSegmentName);
+                                    log.debug("{} openWrite - finished segment={} latency={}.", logPrefix, streamSegmentName, timer.getElapsedMillis());
                                     LoggerHelpers.traceLeave(log, "openWrite", traceId, retValue);
                                     return retValue;
                                 }, executor);
                             }, executor),
-                    executor);
+                    executor)
+                    .handleAsync( (v, ex) -> {
+                        if (null != ex) {
+                            log.warn("{} openWrite - exception segment={} latency={}.", logPrefix, streamSegmentName, timer.getElapsedMillis(), ex);
+                            handleException(streamSegmentName, ex);
```

Comment:
This doesn't look like it's covered by tests. 

Also, won't this effectively swallow the exception? Does `handleException` re-throw it? (if that's the case, you're likely 
 double logging it).

Reply:
same below with the other ones.

Reply:
Also why not use `exceptionally`? That would have saved you the `if` statement.

Reply:
1. On each SLTS methods there are plenty of tests that pass invalid arguments and cause failures etc. Here is one example https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageTests.java#L228

2. Yes handleException rethrows but doesn't log the exception.
3. Make sense to use `exceptionally `so that I save if